sdk/core/az_core/src/az_credential_client_secret.c

```diff
@@ -48,9 +48,11 @@ static AZ_NODISCARD az_result _az_credential_client_secret_request_token(
 az_span const _az_auth_header_name = AZ_SPAN_LITERAL_FROM_STR("authorization");
 
 // This gets called from the http credential policy
-static AZ_NODISCARD az_result _az_credential_client_secret_apply(
+static AZ_NODISCARD az_result _az_credential_client_secret_apply_policy(
+    _az_http_policy* policies,
     az_credential_client_secret* credential,
-    _az_http_request* ref_request)
+    _az_http_request* ref_request,
```

Comment:
Can you please check consistency of parameter names. 
The typedef uses p_policies, p_options, p_request, and p_response.
So, each and every policy function should use the same "p_" prefix and names (except for maybe p_options).
Or, change the "p_" to something else; the important thing is consistency between the typedef and ALL the policy functions.

Reply:
Done :)

Reply:
One thing leads to another, it ended up being touching a lot of files, but on the other hand #496 is mostly done now.